src/site/content/en/blog/native-file-system/index.md

```diff
@@ -184,12 +191,70 @@ async function getNewFileHandle() {
 }
 ```
 
-#### Save changes to the original file
+#### Save changes to disk {: #save-to-disk }
 
 You can find all the code for saving changes to a file in my [text
 editor][text-editor] demo on [GitHub][text-editor-source]. The core file system
 interactions are in [`fs-helpers.js`][text-editor-fs-helper]. At its simpliest,
-the process looks like the code below. I'll walk through each step and explain it.
+the process looks like the code below. I'll walk through each step and explain
+it.
+
+```js
+async function writeFile(fileHandle, contents) {
+  // Create a FileSystemWritableFileStream to write to.
+  const writable = await fileHandle.createWritable();
+  // Write the contents of the file to the stream.
+  await writable.write(contents);
+  // Close the file and write the contents to disk.
+  await writable.close();
+}
+```
+
+Writing data to disk uses a [`FileSystemWritableFileStream`][fs-writablestream],
+essentially a [`WritableStream`][writable-stream]. Create the stream by calling
+`createWritable()` on the file handle object. When `createWritable()` is
+called, Chrome first checks if the user has granted write permission to the
+file. If permission to write hasn't been granted, the browser will prompt
+the user for permission. If permission isn't granted, `createWritable()`
+will throw a `DOMException`, and the app will not be able to write to the
+file. In the text editor, these `DOMException`s are handled in the
+[`saveFile()`][text-editor-app-js] method.
+
+The `write()` method takes a string, which is what we want for a text editor.
+But it can also take a [BufferSource][buffersource], or a [Blob][blob].
+
+For example, you can pipe a stream directly to it:
+
+```js
+async function writeURLToFile(fileHandle, url) {
+  // Create a FileSystemWritableFileStream to write to.
+  const writable = await fileHandle.createWritable();
+  // Make an HTTP request for the contents.
+  const response = await fetch(url);
+  // Stream the response into the file.
+  await response.body.pipeTo(writable);
+  // pipeTo() closes the destination pipe by default, no need to close it.
+}
+```
+
+You can also [`seek()`][spec-seek], or [`truncate`][spec-truncate] within the
+stream to update the file at a specific position, or resize the file.
+
+{% Aside 'caution' %}
+Changes are **not** written to disk until the stream is closed, either by
+calling `close()` or if the stream is closed by the pipe.
```

Comment:
```suggestion
Changes are **not** written to disk until the stream is closed, either when
the script calls `close()` or the stream is closed by the pipe.
```

Reply:
I re-wrote this section before I saw your feedback. I think I've addressed it, but maybe?